src/xsdba/processing.py

```diff
@@ -0,0 +1,872 @@
+        )
+
+    mult, freq, _, _ = parse_offset(xr.infer_freq(time))
+    days_in_year = max_doy[cal]
+    elements_in_year = {"Q": 4, "M": 12, "D": days_in_year, "h": days_in_year * 24}
+    N_in_year = elements_in_year.get(freq, 1) / mult
+    if N_in_year % 1 != 0:
+        raise ValueError(
+            f"Sampling frequency of the data must be Q, M, D or h and evenly divide a year (got {mult}{freq})."
+        )
+
+    return int(N_in_year)
+
+
+@update_xsdba_history
+@harmonize_units(["data", "lower_bound", "upper_bound"])
+def to_additive_space(
+    data: xr.DataArray,
+    lower_bound: str,
+    upper_bound: str | None = None,
+    trans: str = "log",
+):
+    r"""Transform a non-additive variable into an additive space by the means of a log or logit transformation.
+
+    Based on :cite:t:`sdba-alavoine_distinct_2022`.
+
+    Parameters
+    ----------
+    data : xr.DataArray
+        A variable that can't usually be bias-adjusted by additive methods.
+    lower_bound : str
+        The smallest physical value of the variable, excluded, as a Quantity string.
+        The data should only have values strictly larger than this bound.
+    upper_bound : str, optional
+        The largest physical value of the variable, excluded, as a Quantity string.
+        Only relevant for the logit transformation.
+        The data should only have values strictly smaller than this bound.
+    trans : {'log', 'logit'}
+        The transformation to use. See notes.
+
+    Notes
+    -----
+    Given a variable that is not usable in an additive adjustment, this applies a transformation to a space where
+    additive methods are sensible. Given :math:`X` the variable, :math:`b_-` the lower physical bound of that variable
+    and :math:`b_+` the upper physical bound, two transformations are currently implemented to get :math:`Y`,
+    the additive-ready variable. :math:`\ln` is the natural logarithm.
+
+    - `log`
+
+        .. math::
+
+            Y = \ln\left( X - b_- \right)
+
+        Usually used for variables with only a lower bound, like precipitation (`pr`,  `prsn`, etc)
+        and daily temperature range (`dtr`). Both have a lower bound of 0.
+
+    - `logit`
+
+        .. math::
+
+            X' = (X - b_-) / (b_+ - b_-)
+            Y = \ln\left(\frac{X'}{1 - X'} \right)
+
+        Usually used for variables with both a lower and a upper bound, like relative and specific humidity,
+        cloud cover fraction, etc.
+
+    This will thus produce `Infinity` and `NaN` values where :math:`X == b_-` or :math:`X == b_+`.
+    We recommend using :py:func:`jitter_under_thresh` and :py:func:`jitter_over_thresh` to remove those issues.
+
+    See Also
+    --------
+    Related functions
+    from_additive_space : for the inverse transformation.
+    jitter_under_thresh : Remove values exactly equal to the lower bound.
+    jitter_over_thresh : Remove values exactly equal to the upper bound.
```

Comment:
The first word (before the colon) on each line must be a function name.

Reply:
That was a work around because the pre-commit wanted a capital letter, but I didn't want to capitalize function names...